src/main/ui/src/components/Main.react.js

```diff
@@ -40,11 +40,11 @@ export default class Main extends React.Component {
     }
 
     componentWillUnmount() {
-        window.launcher.unregisterLoginRerender(this);
+        window.launcher.unregisterUpdateProfilesRerender(this);
     }
 
-    onLogin(profile) {
-        this.setState({username: profile.name});
+    onUpdateProfiles(profiles) {
+        (profiles.length === 0) ? this.setState({username: false}) : this.setState({username: profiles[0].name});
```

Comment:
Dopplung von `this.setState(...)`
ggf. `this.setState({username: profiles.length === 0 ? false : profiles[0].name})`